Automattic/jetpack
# Jetpack Carousel: Resolve AMP plugin warning

## Proposed changes:

The AMP plugin's `amp_is_available` function cannot be called until the 'wp' hook, but we were calling it during the 'init' hook. Fix that.

### Other information:


## Jetpack product discussion


Fixes #28774
Fixes #26658 (probably)

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Install and connect Jetpack.
* Install and activate the AMP plugin.
* Make sure the Carousel module is enabled.
* Visit the site's frontend.
* Check logs for the lack of the following message in the logs: ``Function amp_is_available was called <strong>incorrectly</strong>. `amp_is_available()` (or `amp_is_request()`, formerly `is_amp_endpoint()`) was called too early and so it will not work properly. WordPress is currently doing the `init` hook. Calling this function before the `wp` action means it will not have access to `WP_Query` and the queried object to determine if it is an AMP response, thus neither the `amp_skip_post()` filter nor the AMP enabled toggle will be considered.``
